### ISSUE TITLE
Allowing assigning the STDOUT result of a task to a variable

### ISSUE THREAD

**Issue by dischello** [go-task/task#178]
Set value of the task to the var:
``` yaml
task-B:
    vars:
        ARG1:
        ARG2:
task-A:
    vars:
      VAR-A: 
        - task-B: 
          vars: {ARG1: "first", ARG2: "second"}
```

it would be nice to have such thing to reuse existing tasks more efficiently or is there way to pass variable/values/data between task?

**Comment by andreynering** [comment]
Hi @dischello,

Sorry, I'm not sure I understood the request. Do you want to assign the result of a task (STDOUT) to a variable?

**Comment by dischello** [comment]
Hi @andreynering,

Sorry for confusion. 
I found solution. Basically i am calling task like this 
``` yaml
task-B:
    vars:
        ARG1:
        ARG2:
task-A:
    vars:
      VAR-A: 
        - task task-B "ARG1=agr1" "ARG2=arg2"
```
worked perfectly fine for me.

**Comment by andreynering** [comment]
@dischello Yeah, that's not ideal, but recursively calling `task` works.

I'll keep this open in order to maybe allow this in the future:

```yaml
vars:
  FOO: {task: foobar}
```

This can be tricky though, since it can in theory allow a recursive dependency (i.e. a var depending on a task and this task depending on the same variable).

For now you can just use [`sh:`](https://taskfile.org/#/usage?id=dynamic-variables).

**Comment by russellseymour** [comment]
@dolfelt where you able to get the `assign` working?

I have been trying to get it working for something I am doing, and have bumped into an issue reading from Stdout.

If you have would you be able to share your code please?

**Comment by orsinium** [comment]
I also have a case where assigning stdout of another command would help. This is the shell command:

```python
sudo docker run -d --publish 8043 --volume ${CREDS}:/srv/http \
    --name secrets-server pierrezemb/gostatic
secrets_ip=$(sudo docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' secrets-server)
sudo docker build ... --add-host=secrets-server:${secrets_ip} ...
```

Here, `secrets_ip` can be set only after the first command is executed and before executing the next one. In the current `task` version, if I translate it as is, the created environment variable is not available in the next command.

**Comment by amancevice** [comment]
Throwing this out there: would it make more sense to attack this problem from the opposite end? eg:

```yaml
tasks:
  get-data:
    cmds:
      - echo "my data"

  use-data:
    vars:
      DATA:
        task: get-data
    cmds:
      - 'echo "{{.DATA}}" | jq | etc'
```

this strategy would conflict with the any-variables experiment, but I think it's cleaner this way.

**Comment by amancevice** [comment]
another workaround I've been playing with using a `scripts` section and YAML anchors to re-use bits of code:

```yaml
scripts:
  AUTO_SCALING_GROUP: &AUTO_SCALING_GROUP
    sh: >-
      aws autoscaling describe-auto-scaling-groups
      | jq -r '.AutoScalingGroups[].AutoScalingGroupName'
      | fzf --prompt '{{.PROMPT | default "Pick Auto Scaling Group > "}}'

tasks:
  list-instances:
    desc: List instance IDs in auto scaling group
    vars:
      PROMPT: "This overrides the prompt below > "
      AUTO_SCALING_GROUP: *AUTO_SCALING_GROUP
    cmds:
      - >-
        aws autoscaling describe-auto-scaling-groups
        --auto-scaling-group-names {{.AUTO_SCALING_GROUP}}
        | jq -r '.AutoScalingGroups[].Instances[].InstanceId
```

**Comment by JonZeolla** [comment]
I was coming here hoping to find something like [this](https://github.com/go-task/task/issues/178#issuecomment-1987345607). Specifically I was hoping to find a way to get this to work:

```
➜ ls
Taskfile.included.yml Taskfile.yml
➜ cat *
---
# https://taskfile.dev

version: '3'

tasks:
  get-vars:
    vars:
      EXAMPLE: 'hit'
---
# https://taskfile.dev

version: '3'

includes:
  example:
    taskfile: Taskfile.included.yml

tasks:
  default:
    cmds:
      - task: example:get-vars
      - echo '{{.EXAMPLE}}'
➜ task -v
task: [/Users/jonzeolla/src/testing] Not found - Using alternative (Taskfile.yml)
task: "default" started
task: "example:get-vars" started
task: "example:get-vars" finished
task: [default] echo ''

task: "default" finished
```

**Comment by dnlopes** [comment]
Has anyone been able to workaround this limitation?

I was trying something like the below, but can't make it work:

```yaml
prompt-user:
  run: always
  silent: true
  interactive: true
  requires:
    vars: [PROMPT]
  cmd: |
    read -rp "{{.PROMPT}}" USER_INPUT
    echo $USER_INPUT

clipboard-encrypt-upload:
  interactive: true
  silent: true
  vars:
    KEY_ID:
      sh: task prompt-user PROMPT="Select the GPG key id to encrypt the data: "
  cmd: echo {{.KEY_ID}}
```

Unfortunately, this does not seem to work; running ```task clipboard-encrypt-upload``` I get the following error:

```
task: Failed to run task "prompt-user": exit status 1
task: Command "task prompt-user PROMPT="Select the GPG key id to encrypt the data:"" failed: exit status 201
```
